### PR TITLE
Home work#9

### PR DESCRIPTION
**Pull request by sergkonhar**
@bohdanpc, @helen-black, проверьте пожалуйста домашнее задание #9

### CODE REVIEW

File: HW#9/Functions.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+	while (inputChar[i] != '\0') {
+
+		if (isalpha(inputChar[i])) {
+			inputChar[i] = toupper(inputChar[i]);
+			i++;
+			continue;
+		}
+		else {
+			inputChar[i] = inputChar[i];
```
Comment (helen-black) [inline HW#9/Functions.cpp]: do you need it?

File: HW#9/Functions.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+#include <iostream>
+
+
+void  stringCharUppercaser(char inputChar[]) {
+
+	int i = 0;
+
+	while (inputChar[i] != '\0') {
+
+		if (isalpha(inputChar[i])) {
```
Comment (helen-black) [inline HW#9/Functions.cpp]: read about isalpha(). do you need this check?

File: HW#9/Functions.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+			continue;
+		}
+		else {
+			inputChar[i] = inputChar[i];
+			i++;
+		}
+
+	}
+
+	std::cout << "Result string: " << inputChar;
```
Comment (helen-black) [inline HW#9/Functions.cpp]: functions  shouldn't speak

File: HW#9/Functions.cpp [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+void  intTransformer(int inputNum[],int length) {
+
+	std::cout << "Result string: ";
+
+	for (int i = 0; i < length; i++) 
+	{
+		if (inputNum[i] > 0) {
+			inputNum[i] *= 2;
+			std::cout << inputNum[i] << " ";
```
Comment (helen-black) [inline HW#9/Functions.cpp]: the same

File: HW#9/Functions.cpp [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+			continue;
```
Comment (helen-black) [inline HW#9/Functions.cpp]: you don't need **continue** here

File: HW#9/Functions.cpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+	
+
+}
+
+bool isPalindrome(char palindromeArray[])
+{
+
+	
+	for (int i = 0; i < strlen(palindromeArray); i++) {
```
Comment (helen-black) [inline HW#9/Functions.cpp]: you can create variable to store strlen(palindromeArray). and use it here and inside of the loop

Comment (helen-black) [inline HW#9/Functions.cpp]: do you need to check the whole string?

File: HW#9/Functions.cpp [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+			else if (i == strlen(firstString) - 1) {
+				return true;
+			}
+		}
+
+	}
+	else {
+		return false;
+	}
+
```
Comment (helen-black) [inline HW#9/Functions.cpp]: you can refactor it:
```c++
if (strlen(firstString) != strlen(secondString)) {
    return false;
}
for (int i = 0; i < strlen(firstString); i++) {
    if (firstString[i] != secondString[i]) {
          return false;
    }
}
return true;
```

File: HW#9/HW9.cpp [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+#include <iostream>
+#include "Functions.h"
+
+
+
+
+
+
+
```
Comment (helen-black) [inline HW#9/HW9.cpp]: a lot of empty spaces

**Review by helen-black — CHANGES_REQUESTED** [review]

**Comment by sergkonhar** [comment]
@bohdanpc, @helen-black, внёс исправления по комментариям

File: HW#9/Functions.cpp [hunk 9]
```diff
@@ -1,113 +1,78 @@
...
 
 bool isPalindrome(char palindromeArray[])
 {
-
-	
-	for (int i = 0; i < strlen(palindromeArray); i++) {
-	
-		if (palindromeArray[i] != palindromeArray[(strlen(palindromeArray) - 1) - i]) {
+	int fullStrLength = strlen(palindromeArray);
+	int halfStrLength = std::ceil(fullStrLength / 2.f)-1;
```
Comment (helen-black) [inline HW#9/Functions.cpp]: int halfStrLength = fullStrLength / 2;

**Pull request merged**